in_toto/runlib.go

```diff
@@ -276,6 +274,12 @@ func InTotoRun(name string, materialPaths []string, productPaths []string,
 		Command:     cmdArgs,
 		Environment: map[string]interface{}{},
 	}
-
+	linkMb.Signatures = []Signature{}
+	// we expect that key has been initialized if it has a valid KeyId
+	if key.KeyId != "" {
```

Comment:
I tried `key != Key{}` actually, it doesn't work with Go. See also: https://stackoverflow.com/questions/28447297/how-to-check-for-an-empty-struct

Key is a struct with another custom type in it, therefore even `if key != Key` will not work properly, because not all fields in it are comparable.

We could add an additional field, as proposed in one of the answers in StackOverflow, or we add an additional function, that manually checks all values for their zero value.

Any other suggestion?

Reply:
Ok I have two other suggestions. We could either do:

`if reflect.DeepEqual(key, Key{})` (this can be very resource wasting)

or we use a new feature from Go 1.13:

`if !reflect.ValueOf(key).IsZero()` this will check if key hasn't been initialized. I guess, this will be better in terms of performance, because we don't need to compare to structs

Reply:
I have just pushed a patch, that uses `IsZero()` if we are unhappy with it, we can look for another solution.
I will leave this comment open, so you can verify it :)

Reply:
Thanks for looking into it. I do like it better than just checking the keyid. What do you think?

Reply:
Yep, looks fine for me, too :+1: